tools/ci_connector_ops/ci_connector_ops/pipelines/utils.py

```diff
@@ -127,9 +127,14 @@ async def get_container_output(container: Container) -> Tuple[str, str]:
     Raises:
         ExecError: If the container exit code is not 0.
     """
-    async with asyncer.create_task_group() as task_group:
-        soon_stdout = task_group.soonify(container.stdout)()
-        soon_stderr = task_group.soonify(container.stderr)()
+    try:
+        async with asyncer.create_task_group() as task_group:
+            soon_stdout = task_group.soonify(container.stdout)()
+            soon_stderr = task_group.soonify(container.stderr)()
+    except anyio.ExceptionGroup as e:
+        if isinstance(e.exceptions[0], ExecError):
+            raise e.exceptions[0]
+        raise
```

Comment:
It's not enough to check the first exception because you could have `ExceptionGroup(..., [QueryError(), ExecError()])`.

With Python 3.11, where [`ExceptionGroup`](https://peps.python.org/pep-0654/#backwards-compatibility) was introduced, you'd use this syntax:
```python
try:
    ...
except* ExecError as eg:
    # now all exceptions in `eg` are of type `ExecError`
    raise next(eg.exceptions)

except* QueryError as eg:
    # no ExecError exceptions in the group, but if there's `QueryError`s use the first 
    raise next(eg.exceptions)
```

There's a backport in earlier versions with the [exceptiongroup](https://pypi.org/project/exceptiongroup/) library.

Reply:
Backport example (not tested!):
```python
from exceptiongroup import ExceptionGroup, catch

def exec_error_handler(eg: ExceptionGroup):
    raise next(eg.exceptions)

async def get_container_output(container: Container) -> Tuple[str, str]:
    with catch({ExecError: exec_error_handler):
        async with asyncer.create_task_group() as task_group:
            soon_stdout = task_group.soonify(container.stdout)()
            soon_stderr = task_group.soonify(container.stderr)()
        return soon_stdout.value, soon_stderr.value
```

Reply:
Thank you! Very elegant syntax in Python 3.11 ! 
Will try the `exceptiongroup` lib for Python 3.10, thanks!

Reply:
I just checked and `anyio` only has support for `exceptiongroup` library in [master](https://github.com/agronholm/anyio/issues/17), which will be released in version 4.0. In case the current implementation isn't compatible, I suggest you just loop until you find the first one:
```python
    except anyio.ExceptionGroup as eg:
        for e in eg.exceptions:
            if isinstance(e, ExecError):
                raise e
        raise
```

Reply:
You can also but that in a context manager to reuse in other places:
```python
@contextlib.contextmanager
def catch_exec_error_group():
    try:
        yield
    except anyio.ExceptionGroup as eg:
        for e in eg.exceptions:
            if isinstance(e, ExecError):
                raise e
        raise

async def get_container_output(container: Container) -> Tuple[str, str]:
    with catch_exec_error_group():
        async with asyncer.create_task_group() as task_group:
            soon_stdout = task_group.soonify(container.stdout)()
            soon_stderr = task_group.soonify(container.stderr)()
        return soon_stdout.value, soon_stderr.value
```

Reply:
@helderco the backport does not work, but your context manager trick is great and works!